### PR TITLE
Add new changes to the account information

### PR DESCRIPTION
**Pull request by priyesh777**
## Addresses:
- Issue: https://github.com/toggle-corp/ifrc-go-meta/issues/155#issue-1741047013

## Changes:
- Add new UI changes to the account information tab of ACCOUNT section

## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/views/Account/AccountInfo/OperationInfoCard/i18n.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "namespace": "accountInformation",
+    "strings": {
+        "operationFollowing": "operation following",
```
Comment (frozenhelium) [inline src/views/Account/AccountInfo/OperationInfoCard/i18n.json]:
```suggestion
        "operationFollowing": "Operations following",
```

File: src/views/Account/AccountInfo/OperationInfoCard/i18n.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+        "operationLastUpdated": "last updated",
```
Comment (frozenhelium) [inline src/views/Account/AccountInfo/OperationInfoCard/i18n.json]:
```suggestion
        "operationLastUpdated": "Last updated",
```

File: src/views/Account/AccountInfo/index.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                            onClick={handleChangePassword}
+                            variant="secondary"
+                        >
+                            Change Password
+                        </Button>
+                    </div>
+                )}
+            >
+                {!!userDetailsResponse && (
+                    <div className={styles.userDetailSection}>
```
Comment (frozenhelium) [inline src/views/Account/AccountInfo/index.tsx]: Use `TextOutput` component for these type of view

File: src/views/Account/AccountInfo/OperationInfoCard/i18n.json (hunk 1 above)
Comment (frozenhelium) [reply src/views/Account/AccountInfo/OperationInfoCard/i18n.json]: This is probably unused

File: src/views/Account/AccountInfo/index.tsx [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+} from '#utils/restRequest';
+import UserContext from '#contexts/user';
+import { Emergency } from '#types/emergency';
+import { User } from '#types/user';
+
+import OperationInfoCard from './OperationInfoCard';
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+interface UserResponse {
```
Comment (frozenhelium) [inline src/views/Account/AccountInfo/index.tsx]: Let's move this to `serverResponse.ts`

File: src/views/Account/AccountInfo/i18n.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+    "namespace": "account",
+    "strings": {
+        "operationFollowing": "operation following",
```
Comment (frozenhelium) [inline src/views/Account/AccountInfo/i18n.json]:
```suggestion
        "operationFollowing": "Operations following",
```

File: src/views/Account/AccountInfo/i18n.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+        "operationLastUpdated": "last updated",
```
Comment (frozenhelium) [inline src/views/Account/AccountInfo/i18n.json]: Probably unused

File: src/views/Account/AccountInfo/i18n.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+        "currentlyFollowing": "currently following operations",
```
Comment (frozenhelium) [inline src/views/Account/AccountInfo/i18n.json]:
```suggestion
        "currentlyFollowing": "Currently following operations",
```

**Review by frozenhelium — CHANGES_REQUESTED** [review]
- Fix all the linting issues
- Remove unused translations
- Rename components appropriately
- Let's use styling from https://www.figma.com/proto/u0XifdJwTEx165x3e0eoP9/IFRC-GO_UI-Prototype?type=design&node-id=673-6133&scaling=scale-down-width&page-id=0%3A4957&starting-point-node-id=0%3A4963&hide-ui=1

**Review by frozenhelium — CHANGES_REQUESTED** [review]
- Use translations
- Use generated typings

**Review by barshathakuri — APPROVED** [review]

**Review by samshara — APPROVED** [review]

File: src/utils/common.ts [hunk 8]
```diff
@@ -215,3 +217,14 @@ export function getDuration(start: Date, end: Date) {
...
+
+export function isWhitelistedEmail<T>(email: string, whitelistedDomains: Maybe<T[]>) {
+    if (!isValidEmail(email) || isNotDefined(whitelistedDomains)) {
+        return false;
+    }
+
+    // Looking for an EXACT match in the domain whitelist
+    // (it finds even if UPPERCASE letters were used)
+    const userMailDomain = email.substring(email.lastIndexOf('@') + 1);
+    return whitelistedDomains.some((domain) => domain === userMailDomain);
```
Comment (tnagorra) [inline src/utils/common.ts]: Let's add a FIXME

File: src/views/AccountInformation/ChangePassword/index.tsx [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+            requiredValidation: requiredStringCondition,
+        },
+        new_password: {
+            required: true,
+            requiredValidation: requiredStringCondition,
+        },
+        confirmNewPassword: {
+            required: true,
+            requiredValidation: requiredStringCondition,
+            forceValue: nullValue,
```
Comment (tnagorra) [inline src/views/AccountInformation/ChangePassword/index.tsx]:
```suggestion
            forceValue: undefinedValue,
```

File: src/views/Register/index.tsx [hunk 10]
```diff
@@ -166,6 +169,8 @@ export function Component() {
 
             setError(formErrors);
             // FIXME: Error message from server is not properly sent
+            // eslint-disable-next-line no-console
+            console.log('Server error::>>', formErrors);
```
Comment (tnagorra) [inline src/views/Register/index.tsx]: Let's remove this console

File: src/views/Register/index.tsx [hunk 11]
```diff
@@ -190,6 +208,10 @@ export function Component() {
         [countriesResponse?.results],
     );
 
+    const whitelistedDomains = whiteListDomainResponse?.results;
+    const validIfrcEmail = formValue.email
+        && isWhitelistedEmail(formValue.email.toLowerCase(), whitelistedDomains);
```
Comment (tnagorra) [inline src/views/Register/index.tsx]: Let's update isWhitelistedEmail

File: src/views/AccountInformation/index.tsx [hunk 12]
```diff
@@ -1,8 +1,218 @@
...
+                            description={meResponse?.profile?.org_type || '--'}
+                        />
+                        <TextOutput
+                            label={strings.departmentLabel}
+                            description={meResponse?.profile?.department || '--'}
+                        />
+                        <TextOutput
+                            label={strings.positionLabel}
+                            description={meResponse?.profile?.position || '--'}
+                        />
```
Comment (tnagorra) [inline src/views/AccountInformation/index.tsx]: Let's update this

**Pull request merged**